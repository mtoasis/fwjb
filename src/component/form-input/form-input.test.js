import React from 'react';
import FormInput from './form-input.component';
import { shallow } from 'enzyme'


describe('Form-input', () => {
 
    let wrapper;
    let mockHandleChange = jest.fn()

    beforeEach(() => {
        const mockProps = {
            label: 'name',
            value: 'Julian',
            handleChange: mockHandleChange
        };
        wrapper = shallow(<FormInput {...mockProps} />)
    })


    it('should render the form-input component', ()=>{
        expect(wrapper).toMatchSnapshot();
    })

    it('should call handleChange when input changes', ()=>{
        wrapper.find('FormInputContainer').simulate('change');
        expect(mockHandleChange).toHaveBeenCalled();
    })    

 
    it('should show the label if there is a lable', ()=>{
        expect(wrapper.exists('FormInputLabelContainer')).toBe(true);
    })

    it('should not show the label if there is no lable', ()=>{
        const mockNewProps = {
            label: '',
            value: 'Julian',
            handleChange: mockHandleChange
        };
        const newWrapper = shallow(<FormInput {...mockNewProps} />)
        expect(newWrapper.exists('FormInputLabelContainer')).toBe(false);
    })

})