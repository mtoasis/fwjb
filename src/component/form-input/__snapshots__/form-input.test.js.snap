// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form-input should render the form-input component 1`] = `
<styled.div>
  <FormInputContainer
    onChange={[MockFunction]}
    value="Julian"
  />
  <FormInputLabelContainer
    value="Julian"
  >
    name
  </FormInputLabelContainer>
</styled.div>
`;
